[EDNA-115] Include primary IC50 into ExperimentResp

gromakovsky opened the pull request:
## Description

Problem: in order to show Dashboard table we query ExperimentsResp
    first and then for each experiment in the response we query its primary
    sub-experiment one-by-one. It's a bit inefficient and we can easily
    decrease number of HTTP endpoint calls.
    
Solution:
1. Add a new field to `ExperimentResp` that holds IC50 computed for the primary sub-experiment. Experiment getter was already querying all sub-experiments, so adding this data to response is pretty straightforward.
2. Update frontend types with this new field and update logic accordingly.

## Related issue(s)

<!--
- Short description of how the PR relates to the issue, including an issue link.
For example
- Fixed #100500 by adding lenses to exported items

Write 'None' if there are no related issues (which is discouraged).
-->

https://issues.serokell.io/issue/EDNA-115

## :white_check_mark: Checklist for your Pull Request

<!--
Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

If you don't set a checkmark (e. g. don't add a test for new functionality),
you must be able to justify that.
-->

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

gromakovsky committed:
Add primary IC50 to ExperimentResp

Problem: in order to show Dashboard table we query ExperimentsResp
first and then for each experiment in the response we query its primary
sub-experiment one-by-one. It's a bit inefficient and we can easily
decrease number of HTTP endpoint calls.

Solution: add a new field to ExperimentResp that holds IC50 computed
for the primary sub-experiment. Experiment getter was already querying
all sub-experiments, so adding this data to response is pretty
staightforward. We extend an existing test to check `erPrimaryIC50` sanity.

gromakovsky committed:
Return no mean IC50 when there are no values

Problem: `computeMeanIC50` returns 0 if there are no experiments
with successfully computed IC50. But we already support the ability
to return no mean IC50 and for no successful experiments it seems
to make more sense than 0, because 0 may mean that all IC50s are 0
for example.

Solution: return `Nothing` if there are no successful IC50s instead
of 0.

gromakovsky committed:
Do not fetch each sub-experiment to show Dashboard table

Problem: in order to show Dashboard table we fetch sub-experiments
one-by-one, so there are a lot of requests as the number of experiments
grows. We fetch them just to get primary IC50.

Solution: now that primary IC50 is provided by the backend as part
of ExperimentResp, we use this value instead of fetching SubExperiment.
Experiment and ExperimentDto types were extended with a new field.
`ResultDto` now supports the ability to store just one number.

pva701 commented on frontend/src/api/types.ts:
```diff
@@ -26,12 +26,12 @@ export type ParsedExcelDto = {
 };
 
-export type ResultDto =
+export type ResultDto<T = number[]> =
```
:POG: ARE YOU TYPESCRIPT EXPERT?

gromakovsky replied:
Just looked at `interface Dto<T, Extra = undefined> {` and did something similar :slightly_smiling_face:

pva701 committed:
Don't request experiment's subexperiments until a user opens experiment table collapsible

Problem: we request all subexperiments for every experiment in the
experiment table on Dashboard page, though, nobody will ever open them.
This happens because we don't check if a collapsible is visible before
rendering 'ExperimentsCollapse' component.

Solution: add check if experiment's collapse is visible
before rendering a collapsible. It will cause a spinner
for first time we open it.
